tests/lib/pre2017.test.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Pre2017 Library Test
+ */
+
+const fs = require('fs')
+const pre2017 = require('../../lib/pre2017')
+
+describe('lib.pre2017', () => {
+  const basedir = `${__dirname}/../__data__/pre2017html`
+  let pre2017Response
+
+  beforeEach(async () => {
+    pre2017Response = await pre2017.get()
+  })
+
+  test('the json results for correct data', async () => {
+    const fetchingExpectedData = pre2017.PRE_URL.map(async url => {
+      const year = url.split('/').pop()
+      return await pre2017.scrape(
+        fs.readFileSync(`${basedir}/${year}.html`).toString(),
```

Comment:
It is **really** *really* weird to have the html but still fetching it via `pre2017.get()` and `get()` is calling `scrape` again internally, so what's the point of this test?

Let's go with this approach and see if you understand.
The function job is to generate a javascript object,
we need to make sure that the object is correct,
for that, you have to have the actual object first,
run the function once (outside the test; manually running it by yourself),
store the return value somewhere (json file?)

After that, you create a test, calling `pre2017`,
and test that it returns the same object that you have before.

Reply:
It's look like what i've done before

Reply:
Your previous one is only testing if the json file exist?

We need it to be correct, not just "existed"; pardon me if you were already asserting it to be correct before.

Reply:
```js

Reply:
Like this?
```js
/**
 * Pre2017 Library Test
 */

const fs = require('fs')
const pre2017 = require('../../lib/pre2017')

describe('lib.pre2017', () => {

  it('should have the correct contents',async () => {
    const expectedData = JSON.parse(
      fs.readFileSync(`${__dirname}/../__data__/pre2017.json`)
    )
    const pre2017Response = await pre2017()
    const pre2017json = pre2017Response.map(data => data.json)

    expect(pre2017json).toEqual(expectedData)
  })

})
```

Reply:
Yep